packages/sdk/src/browser-wallets/magiceden/index.ts

```diff
@@ -19,46 +20,50 @@ export interface MagicEdenWindow extends Window {
   BitcoinProvider?: MagicEdenBitcoinProvider;
 }
 
+export interface MagicEdenWallet extends Wallet {
+  features: {
+    "sats-connect:": {
+      provider: Promise<BitcoinProvider>;
+    };
+  };
+}
+
 /**
  * Checks if the MagicEden Wallet extension is installed.
  *
  * @returns `true` if installed, `false` otherwise.
  * @throws {OrditSDKError} Function is called outside a browser without `window` object
  */
-function isInstalled(): boolean {
+async function isInstalled(
+  getMeProvider: () => Promise<BitcoinProvider>,
+): Promise<boolean> {
   if (typeof window === "undefined") {
     throw new OrditSDKError("Cannot call this function outside a browser");
   }
 
+  const magicEdenProvider = await getMeProvider();
+
   return (
-    typeof (window as MagicEdenWindow).BitcoinProvider?.isMagicEden !==
+    typeof (magicEdenProvider as MagicEdenBitcoinProvider).isMagicEden !==
     "undefined"
   );
 }
 
-async function getMagicEdenWalletProvider(): Promise<BitcoinProvider> {
-  if (!isInstalled()) {
-    throw new BrowserWalletNotInstalledError(
-      "Magic Eden not installed or set as prioritised wallet.",
-    );
-  }
-
-  return window.BitcoinProvider!;
-}
-
 async function getAddresses(
+  getMeProvider: () => Promise<BitcoinProvider>,
   network: BrowserWalletNetwork = "mainnet",
 ): Promise<WalletAddress[]> {
-  if (!isInstalled()) {
+  if (!isInstalled(getMeProvider)) {
     throw new BrowserWalletNotInstalledError(
       "Magic Eden not installed or set as prioritised wallet.",
```

Comment:
as we're required to support ME without setting as prioritised wallet, the error message will be need to changed:

```suggestion
      "Magic Eden not installed.",
```